src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -129,6 +129,13 @@ namespace winrt::TerminalApp::implementation
             p.SetActionMap(_settings.ActionMap());
         }
 
+        // If the active LLMProvider changed, make sure we reinitialize the provider
+        const auto newProviderType = _settings.GlobalSettings().AIInfo().ActiveProvider();
+        if (_lmProvider && (newProviderType != _currentProvider))
```

Comment:
Why does this depend on the existence of `_lmProvider`?

Reply:
We only want to reinitialize the `_lmProvider` if there actually was one before the active provider changed. If the active provider changed and there wasn't a previous `_lmProvider` we're alright, it will get initialized when the chat gets opened. This is to handle the scenario where the user was using the chat, and then went to their settings and changed the active provider.

Reply:
IMO, I think this may be worth adding as a comment, because it's not immediately obvious to me that `_lmProvider` is used as a hint as to whether the chat is currently open. And that if the chat is currently closed, then the provider gets refreshed once it opens.